force-app/main/default/classes/DMLOperations.cls

```diff
@@ -148,6 +174,41 @@ public with sharing class DMLOperations {
         // Loop through each Opportunity name
             // Create an Opportunity with the given name
             // Add the Opportunity to the list
+        /**** ORIGINAL CODE
+        Account acc = new Account(Name = accountName);
+        upsert acc;
+        
+        List<Opportunity> newOpps = new List<Opportunity>();
+        for (String oppName : oppNames) {
+            Opportunity newOpp = new Opportunity(Name = oppName, AccountId = acc.Id, StageName = 'Prospecting', CloseDate = Date.today().addMonths(1));
+            newOpps.add(newOpp);
+        }
+        upsert newOpps;
+        ****/
+        
+        // Per Justin's suggestion, refactoring method to check for account/opportunities first
+        // Check if account exists, if not insert
+        Account account = new Account();
+        List<Account> accounts = [SELECT Id, Name FROM Account WHERE Name = :accountName];
+        
+        if (accounts.isEmpty()) {
+            account = new Account(Name = accountName);
+            insert account;
+        } else {
+            // Grab first account w/this name
+            account = accounts[0];
+        }
+
+        // Get existing opps by name; if not exists then insert
+        List<Opportunity> oppsNew = new List<Opportunity>();
+        List<Opportunity> oppsExisting = [SELECT Id, Name FROM Opportunity WHERE Name IN :oppNames];
+
+        for (String oppName : oppNames) {
+            if (!oppsExisting.toString().contains(oppName)) {
+                oppsNew.add(new Opportunity(Name = oppName, AccountId = account.Id, StageName = 'Prospecting', CloseDate = Date.today().addMonths(1)));
```

Comment:
Based on the method instructions, we'd also want to do this for existing Opportunities (update fields) not just when we create new opportunities. (But to be honest, the directions are not super clear what to do with existing opps. At the very least, we'd want to make sure they were related to the account whose name was provided as a parameter.)
* Question 8
     * Insert or updates opportunities from list of names.
     * Associate the Opportunities with account that has the name accountName.
     * There should be an Opportunity for each name provided in the list.